packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -95,6 +116,13 @@ interface ISerializerEvent extends IEvent {
 	(event: "saved", listener: (dirty: boolean) => void): void;
 }
 
+/**
+ * Helper class to manage the state of the container needed for proper serialization.
+ *
+ * It holds the pendingLocalState the container was rehydrated from (if any),
+ * as well as the snapshot to be used for serialization.
+ * It also keeps track of container dirty state and which local ops have been processed
```

Comment:
This part is a bit fuzzy for me still, open to suggestions on rewording/elaborating.